pkg/tfvars/tfvars.go

```diff
@@ -57,7 +57,6 @@ func TFVars(cfg *types.InstallConfig, bootstrapIgn, masterIgn, workerIgn string)
 			if m.Platform.AWS != nil {
 				config.AWS.Master = aws.Master{
 					EC2Type:     m.Platform.AWS.InstanceType,
-					IAMRoleName: m.Platform.AWS.IAMRoleName,
```

Comment:
i think this forces terraform to create one for you.

Reply:
> i think this forces terraform to create one for you.

I haven't tracked that down, but we'll also want to remove/prune some of:

```
$ git grep iam_role origin/pr/697
origin/pr/697:data/data/aws/bootstrap/main.tf:  role = "${var.iam_role == "" ?
origin/pr/697:data/data/aws/bootstrap/main.tf:    join("|", aws_iam_role.bootstrap.*.name) :
origin/pr/697:data/data/aws/bootstrap/main.tf:    join("|", data.aws_iam_role.bootstrap.*.name)
origin/pr/697:data/data/aws/bootstrap/main.tf:data "aws_iam_role" "bootstrap" {
origin/pr/697:data/data/aws/bootstrap/main.tf:  count = "${var.iam_role == "" ? 0 : 1}"
origin/pr/697:data/data/aws/bootstrap/main.tf:  name  = "${var.iam_role}"
origin/pr/697:data/data/aws/bootstrap/main.tf:resource "aws_iam_role" "bootstrap" {
origin/pr/697:data/data/aws/bootstrap/main.tf:  count = "${var.iam_role == "" ? 1 : 0}"
origin/pr/697:data/data/aws/bootstrap/main.tf:resource "aws_iam_role_policy" "bootstrap" {
origin/pr/697:data/data/aws/bootstrap/main.tf:  count = "${var.iam_role == "" ? 1 : 0}"
origin/pr/697:data/data/aws/bootstrap/main.tf:  role  = "${aws_iam_role.bootstrap.id}"
origin/pr/697:data/data/aws/bootstrap/variables.tf:variable "iam_role" {
origin/pr/697:data/data/aws/iam/main.tf:  role = "${var.worker_iam_role == "" ?
origin/pr/697:data/data/aws/iam/main.tf:    join("|", aws_iam_role.worker_role.*.name) :
origin/pr/697:data/data/aws/iam/main.tf:    join("|", data.aws_iam_role.worker_role.*.name)
origin/pr/697:data/data/aws/iam/main.tf:data "aws_iam_role" "worker_role" {
origin/pr/697:data/data/aws/iam/main.tf:  count = "${var.worker_iam_role == "" ? 0 : 1}"
origin/pr/697:data/data/aws/iam/main.tf:  name  = "${var.worker_iam_role}"
origin/pr/697:data/data/aws/iam/main.tf:resource "aws_iam_role" "worker_role" {
origin/pr/697:data/data/aws/iam/main.tf:  count = "${var.worker_iam_role == "" ? 1 : 0}"
origin/pr/697:data/data/aws/iam/main.tf:resource "aws_iam_role_policy" "worker_policy" {
origin/pr/697:data/data/aws/iam/main.tf:  count = "${var.worker_iam_role == "" ? 1 : 0}"
origin/pr/697:data/data/aws/iam/main.tf:  role  = "${aws_iam_role.worker_role.id}"
origin/pr/697:data/data/aws/iam/variables.tf:variable "worker_iam_role" {
origin/pr/697:data/data/aws/main.tf:  iam_role                    = "${var.tectonic_aws_master_iam_role_name}"
origin/pr/697:data/data/aws/main.tf:  master_iam_role          = "${var.tectonic_aws_master_iam_role_name}"
origin/pr/697:data/data/aws/main.tf:  worker_iam_role = "${var.tectonic_aws_worker_iam_role_name}"
origin/pr/697:data/data/aws/master/main.tf:  role = "${var.master_iam_role == "" ?
origin/pr/697:data/data/aws/master/main.tf:    join("|", aws_iam_role.master_role.*.name) :
origin/pr/697:data/data/aws/master/main.tf:    join("|", data.aws_iam_role.master_role.*.name)
origin/pr/697:data/data/aws/master/main.tf:data "aws_iam_role" "master_role" {
origin/pr/697:data/data/aws/master/main.tf:  count = "${var.master_iam_role == "" ? 0 : 1}"
origin/pr/697:data/data/aws/master/main.tf:  name  = "${var.master_iam_role}"
origin/pr/697:data/data/aws/master/main.tf:resource "aws_iam_role" "master_role" {
origin/pr/697:data/data/aws/master/main.tf:  count = "${var.master_iam_role == "" ? 1 : 0}"
origin/pr/697:data/data/aws/master/main.tf:resource "aws_iam_role_policy" "master_policy" {
origin/pr/697:data/data/aws/master/main.tf:  count = "${var.master_iam_role == "" ? 1 : 0}"
origin/pr/697:data/data/aws/master/main.tf:  role  = "${aws_iam_role.master_role.id}"
origin/pr/697:data/data/aws/master/variables.tf:variable "master_iam_role" {
origin/pr/697:data/data/aws/variables-aws.tf:variable "tectonic_aws_master_iam_role_name" {
origin/pr/697:data/data/aws/variables-aws.tf:variable "tectonic_aws_worker_iam_role_name" {
origin/pr/697:pkg/tfvars/aws/aws.go:    IAMRoleName      string            `json:"tectonic_aws_master_iam_role_name,omitempty" yaml:"iamRoleName,omitempty"`
origin/pr/697:pkg/tfvars/aws/aws.go:    IAMRoleName      string            `json:"tectonic_aws_worker_iam_role_name,omitempty" yaml:"iamRoleName,omitempty"`
```

as well.  I'm not sure about the bootstrap role.  Maybe we should keep that?

Reply:
Also should prune:

```console
$ git grep -i iamrole origin/pr/697 -- pkg
origin/pr/697:pkg/tfvars/aws/aws.go:    IAMRoleName      string            `json:"tectonic_aws_master_iam_role_name,omitempty" yaml:"iamRoleName,omitempty"`
origin/pr/697:pkg/tfvars/aws/aws.go:    IAMRoleName      string            `json:"tectonic_aws_worker_iam_role_name,omitempty" yaml:"iamRoleName,omitempty"`
origin/pr/697:pkg/types/aws/machinepool.go:     // IAMRoleName defines the IAM role associated
origin/pr/697:pkg/types/aws/machinepool.go:     IAMRoleName string `json:"iamRoleName"`
origin/pr/697:pkg/types/aws/machinepool.go:     if required.IAMRoleName != "" {
origin/pr/697:pkg/types/aws/machinepool.go:             a.IAMRoleName = required.IAMRoleName
```

and related.